src/publisher.ts

```diff
@@ -182,6 +187,12 @@ export class StreamPublisher implements Publisher {
   }
 
   private async enqueue(publishRequestMessage: PublishRequestMessage) {
+    if (this.filter) {
+      publishRequestMessage.filterValue = this.filter(publishRequestMessage.message)
+    }
+    if (!this.client.isFilteringEnabled && this.filter) {
```

Comment:
I think the error should be thrown only if `this.filter` is defined (let's use the short-circuit evaluation)

```suggestion
    if (this.filter && !this.client.isFilteringEnabled) {
```